rnd/nextgenautogpt/__main__.py

```diff
@@ -0,0 +1,39 @@
+import multiprocessing as mp
+from typing import Any
+
+import nextgenautogpt.manager.manager as mod_manager
+import nextgenautogpt.server.server as mod_server
+
+
+def main() -> None:
+    # Create queues/pipes for communication
+    server_to_manager: mp.Queue[Any] = mp.Queue()
+    manager_to_server: mp.Queue[Any] = mp.Queue()
```

Comment:
Also, should we add the queue abstraction to hide the implementation detail?

e.g: 
class queue:

   def  publish_event(name, content):
       pass

    def consume_event(name) -> content:
      pass
```

Reply:
The intent of this PR is to give a program sketch so we can all work together on it.

Reply:
Just read the whole PR again, I think the need of the Queue abstraction is even higher since we will be sharing the queue between the server and producer.

This is what I imagined:

```
class Message:
    pass

class MessageSubscriber:
     async def read_next_event() -> Message:
          pass

class MessageQueue:
     async def publish(event_name: str, msg: Message):
         pass

     async def consume_event(event_name: str) -> Message:
         pass
    
     async def subscribe_event(event_name: str) -> MesageSubscriber:
         pass
```

or something along the line, then you can use the the zeromq python bindings or any other queue library under the hood.

Reply:
Yep agreed